File: connector-sdk/src/models/enums.ts
// Domain enumerations — wire values mirror the Swift agent's VGCore/Enums.swift
// 1:1 (snake_case / PascalCase raw strings), so JSON envelopes stay stable
// across the Rust engine, the Swift agent, this SDK, and the dashboard.
//
// Source of truth: agent/Sources/VGCore/Enums.swift

import { z } from 'zod';

/** Enforcement action returned by the engine. */
export const ActionSchema = z.enum(['allow', 'warn', 'block']);
export type Action = z.infer<typeof ActionSchema>;

/** Aggregate risk level of an evaluated prompt. */
export const RiskLevelSchema = z.enum(['low', 'medium', 'high', 'critical']);
export type RiskLevel = z.infer<typeof RiskLevelSchema>;

/** Severity of a rule or finding. */
export const SeveritySchema = z.enum(['low', 'medium', 'high', 'critical']);
export type Severity = z.infer<typeof SeveritySchema>;

/** Data classification of prompt content. */
export const ClassificationSchema = z.enum([
  'public',
  'internal',
  'confidential',
  'restricted',
]);
export type Classification = z.infer<typeof ClassificationSchema>;

/**
 * Detector category — the 24 policy categories plus the legacy
 * `classification` derivation category. Wire values are snake_case.
 */
export const CategorySchema = z.enum([
  'secret',
  'pii',
  'source_code',
  'classification',
  'company_confidential',
  'financial',
  'intellectual_property',
  'usage_policy',
  'prompt_injection',
  'sensitive_document',
  'customer_data',
  'compliance',
  'keyword',
  'file_policy',
  'image_policy',
  'ai_classification',
  'destructive_command',
  'legal',
  'medical',
  'hr',
  'security',
  'research_development',
  'communication',
  'procurement',
  'government',
]);
export type Category = z.infer<typeof CategorySchema>;

/** Origin surface of a prompt. */
export const SourceSchema = z.enum(['browser', 'ide', 'cli', 'api']);
export type Source = z.infer<typeof SourceSchema>;

/** RBAC role of the acting user. */
export const RoleSchema = z.enum([
  'super_admin',
  'security_admin',
  'auditor',
  'manager',
  'user',
]);
export type Role = z.infer<typeof RoleSchema>;

/**
 * Audit event types (EVENT_MODEL.md). Raw values are the canonical PascalCase
 * names persisted in envelopes and uploaded to the backend.
 */
export const EventTypeSchema = z.enum([
  'PromptSubmitted',
  'PolicyEvaluated',
  'PromptAllowed',
  'PromptWarned',
  'WarningAccepted',
  'WarningRejected',
  'PromptBlocked',
  'PolicyViolation',
  'UploadSuccess',
  'UploadFailure',
  'AgentStarted',
  'PolicyUpdated',
]);
export type EventType = z.infer<typeof EventTypeSchema>;

/** The primary event type that corresponds to a decision's action. */
export function primaryEventType(action: Action): EventType {
  switch (action) {
    case 'allow':
      return 'PromptAllowed';
    case 'warn':
      return 'PromptWarned';
    case 'block':
      return 'PromptBlocked';
  }
}

File: agent/Sources/VGCore/Enums.swift
// Domain enumerations for the VGuardrail agent.
//
// Raw values are the canonical wire strings shared with the policy engine and the
// backend (snake_case), so JSON envelopes are stable across the Rust engine, this
// agent, and the dashboard.

import Foundation

/// Enforcement action returned by the engine.
public enum Action: String, Codable, Sendable, CaseIterable {
    case allow
    case warn
    case block
}

/// Aggregate risk level of an evaluated prompt.
public enum RiskLevel: String, Codable, Sendable, CaseIterable {
    case low
    case medium
    case high
    case critical
}

/// Severity of a rule or finding.
public enum Severity: String, Codable, Sendable, CaseIterable {
    case low
    case medium
    case high
    case critical
}

/// Data classification of prompt content.
public enum Classification: String, Codable, Sendable, CaseIterable {
    case `public`
    case `internal`
    case confidential
    case restricted
}

/// Detector category — the 24 policy categories plus the legacy
/// `classification` derivation category.
public enum Category: String, Codable, Sendable, CaseIterable {
    case secret
    case pii
    case sourceCode = "source_code"
    case classification
    case companyConfidential = "company_confidential"
    case financial
    case intellectualProperty = "intellectual_property"
    case usagePolicy = "usage_policy"
    case promptInjection = "prompt_injection"
    case sensitiveDocument = "sensitive_document"
    case customerData = "customer_data"
    case compliance
    case keyword
    case filePolicy = "file_policy"
    case imagePolicy = "image_policy"
    case aiClassification = "ai_classification"
    case destructiveCommand = "destructive_command"
    case legal
    case medical
    case hr
    case security
    case researchDevelopment = "research_development"
    case communication
    case procurement
    case government
}

/// Origin surface of a prompt.
public enum Source: String, Codable, Sendable, CaseIterable {
    case browser
    case ide
    case cli
    case api
}

/// RBAC role of the acting user.
public enum Role: String, Codable, Sendable, CaseIterable {
    case superAdmin = "super_admin"
    case securityAdmin = "security_admin"
    case auditor
    case manager
    case user
}

/// Audit event types (EVENT_MODEL.md). Raw values are the canonical PascalCase
/// names persisted in envelopes and uploaded to the backend.
public enum EventType: String, Codable, Sendable, CaseIterable {
    case promptSubmitted = "PromptSubmitted"
    case policyEvaluated = "PolicyEvaluated"
    case promptAllowed = "PromptAllowed"
    case promptWarned = "PromptWarned"
    case warningAccepted = "WarningAccepted"
    case warningRejected = "WarningRejected"
    case promptBlocked = "PromptBlocked"
    case policyViolation = "PolicyViolation"
    case uploadSuccess = "UploadSuccess"
    case uploadFailure = "UploadFailure"
    case agentStarted = "AgentStarted"
    case policyUpdated = "PolicyUpdated"

    /// The primary event type that corresponds to a decision's action.
    public static func primary(for action: Action) -> EventType {
        switch action {
        case .allow: return .promptAllowed
        case .warn: return .promptWarned
        case .block: return .promptBlocked
        }
    }
}
